src/data/proxy/UserProfileStore.java

```diff
@@ -1,44 +1,28 @@
 package data.proxy;
 
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.function.Predicate;
 
-import data.proxy.utils.FunctionalUtils;
 import data.structure.UserProfile;
 
 /**
  * UserProfileStore manages access to the stored user profiles.
  */
-public class UserProfileStore {
-    private Map<String, UserProfile> userProfiles;
-    
-    /**
-     * Basic default constructor for UserProfileStore.
-     */
-    public UserProfileStore() {
-        userProfiles = new HashMap<String, UserProfile>();
-    }
+public abstract interface UserProfileStore {
```

Comment:
Generally, I stay away from that type of convention, similarly to how many of Java's native libraries are implemented (eg. List, Map, Set, etc.). When the class is such that there will definitely be multiple possible implementations, such as any storage interface, I think that it can be reasonably inferred, but at the end of the day, it's still just a coding standard that I choose to stick with.